stacker/actions/diff.py

```diff
@@ -0,0 +1,146 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _diff_parameters(self, old_params, new_params):
+        """Compares the old vs. new parameters and returns a 'diff'
+
+        Args:
+            old_params(dict): old paramters
+            new_params(dict): new parameters
+
+        Returns:
+            int: number of changed entries
+            string: "diff" of all parameters and how they changed
+        """
+        output = []
+        count = 0
+        for k, v in old_params.iteritems():
+            new_val = new_params.get(k, "_Undef_")
+            if str(v) != str(new_val):
+                output.append("CHANGED   %s: %s => %s\n" % (k, v, new_val))
+                count += 1
+            else:
+                output.append("NO CHANGE %s: %s\n" % (k, v))
+
+        for k, v in new_params.iteritems():
+            if not k in old_params:
+                output.append("NEW       %s: %s\n" % (k, v))
+                count += 1
+
+        return [count, "".join(output)]
```

Comment:
Not sure I understand?  output is the diff.  I suppose that I could just suppress the "NO CHANGE" lines, but decided I kinda liked seeing all the parameters if the parameters or template itself changed.